src/Scripting/Core/Hosting/ObjectFormatter/CommonTypeNameFormatter.cs

```diff
@@ -93,7 +93,7 @@ public virtual string FormatTypeArguments(Type[] typeArguments, CommonTypeNameFo
 
             if (typeArguments.Length == 0)
             {
-                return "";
+                throw new ArgumentException(nameof(typeArguments));
```

Comment:
@davkean, yes that was intended to be the paramName.  I naively assumed that the parameter order matched `ArgumentNullException`.

@tmat, it was public when I originally wrote the validation, but I can switch to asserts now.